.github/workflows/build.yml

```diff
@@ -94,58 +94,77 @@ jobs:
         push: true
         tags: signadot/hotrod-route:${{ github.sha }}
 
-  update_signadot_customer:
+  update_signadot:
     needs: [build, changes]
-    if: ${{ needs.changes.outputs.customer == 'true' }}
     runs-on: ubuntu-latest
     steps:
-    - name: curl
-      uses: wei/curl@v1.1.1
+    - name: call signadot API
+      id: upsert_workspace
+      run: |      
+         imageReplacements=()
+         if [[ ${{ needs.changes.outputs.customer }} == 'true' ]]
+         then
+           imageReplacements+=('{"name": "signadot/hotrod-customer", "newTag": "${{ github.sha }}"}')
+         fi
+         
+         if [[ ${{ needs.changes.outputs.driver }} == 'true' ]]
+         then
+           imageReplacements+=('{"name": "signadot/hotrod-driver", "newTag": "${{ github.sha }}"}')
+         fi
+         
+         if [[ ${{ needs.changes.outputs.frontend }} == 'true' ]]
+         then
+           imageReplacements+=('{"name": "signadot/hotrod-frontend", "newTag": "${{ github.sha }}"}')
+         fi
+         
+         if [[ ${{ needs.changes.outputs.route }} == 'true' ]]
+         then
+           imageReplacements+=('{"name": "signadot/hotrod-route", "newTag": "${{ github.sha }}"}')
+         fi
+         
+         printf -v joined '%s,' "${imageReplacements[@]}"
+         if [ ${#imageReplacements[@]} -eq 0 ]; then
+              echo "no update required"
+              exit 1
```

Comment:
This seems to make the whole action get marked as failed, which looks like a problem. Should we `exit 0`? Or maybe we should even continue on to upsert an empty workspace so that previous images for this PR get removed from the workspace. There should be almost no cost to an empty workspace since it doesn't have any cloned workloads.

Reply:
Hmm, I had it that way before. It does look jarring to have the entire CI job fail here, so, will switch it back to `exit 0`.